src/platforms/atomic-kms/server/display_helpers.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_GRAPHICS_ATOMIC_DISPLAY_HELPERS_H_
+#define MIR_GRAPHICS_ATOMIC_DISPLAY_HELPERS_H_
+
+#include "mir/udev/wrapper.h"
+#include "mir/fd.h"
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic warning "-Wall"
+#include <gbm.h>
+#pragma GCC diagnostic pop
```

Comment:
I'm curious: most places we include `<gbm.h>` we don't disable warnings. Why is this occasion different?